test/zimwriterfs-zimcreatorfs.cpp

```diff
@@ -75,18 +75,63 @@ TEST(ZimCreatorFSTest, MinimalZim)
   zimCreator.startZimCreation(out.path());
   zimCreator.visitDirectory(directoryPath);
 
-  std::shared_ptr<zim::writer::Article> redirect_article(new RedirectArticle(&zimCreator, 'A', "index.html", "Start page", zim::writer::Url("A/hello.html")));
+  std::shared_ptr<zim::writer::Article> redirect_article(
+        new RedirectArticle(&zimCreator, 'A', "index.html", "Start page",
+                            zim::writer::Url("A/hello.html")));
   zimCreator.addArticle(redirect_article);
 
   zimCreator.finishZimCreation();
 
-  // verify the created .zim file with 'zimdump'
+
+  // VERIFY the created .zim file with 'zimdump'
   zim::File zimfile(out.path());
   EXPECT_EQ(zimfile.getCountArticles(), 4);
 
   zim::Article a1 = zimfile.getArticle('A', "index.html");
   EXPECT_TRUE(a1.isRedirect());
 
+  uint32_t main_index = zimfile.getFileheader().getMainPage();
+  EXPECT_EQ(a1.getIndex(), main_index);
+
   zim::Article a2 = a1.getRedirectArticle();
   EXPECT_EQ(a2.getTitle(), "HTML title tag content");
 }
+
+TEST(ZimCreatorFSTest, SymlinkShouldCreateRedirectArticleEntry)
+{
+  LibMagicInit libmagic;
+
+  std::string directoryPath = "data/with-symlink";
+  ZimCreatorFS zimCreator(directoryPath, "hello.html", false, false);
+
+  TempFile out("with-symlink.zim");
+
+  zimCreator.startZimCreation(out.path());
+  zimCreator.visitDirectory(directoryPath);
+
+  zimCreator.addArticle(
+        std::make_shared<MetadataFaviconArticle>(zim::writer::Url('I', "link.jpeg")));
```

Comment:
Why do we need this article ?

Reply:
It's 'favicon' metadata which usually added in the same way from main() in zimwriterfs.cpp
It's equivalent of calling `zimwriterfs -f link.jpeg`:

![Screenshot from 2020-09-15 12-43-27](https://user-images.githubusercontent.com/5078434/93170433-6ceb6800-f751-11ea-9b0a-ccfd4c093e20.png)

Reply:
Yes, but we don't need it. We can create a zim without the favicon. And we don't test it.
There is no need to create a zim file with it (and store the file in the git history).